coretex/entities/conversion/base_converter.py

```diff
@@ -64,12 +64,11 @@ class BaseConverter(ABC):
     """
 
     def __init__(self, datasetName: str, projectId: int, datasetPath: str) -> None:
-        dataset: Optional[ImageDatasetType] = ImageDataset.createDataset(datasetName, projectId)
-        if dataset is None:
-            raise ValueError(">> [Coretex] Failed to create dataset")
+        dataset: ImageDatasetType = ImageDataset.createDataset(datasetName, projectId)
 
         self._dataset: Final = dataset
         self._datasetPath: Final = datasetPath
+        dataset.finalizeState()
```

Comment:
This will fail, dataset is getting samples added to it in `_saveImageAnnotationPair` function. State needs to be set to final after all samples have been added to it.